src/main/java/org/eclipsefoundation/marketplace/helper/StreamHelper.java

```diff
@@ -0,0 +1,49 @@
+/* Copyright (c) 2019 Eclipse Foundation and others.
+ * This program and the accompanying materials are made available
+ * under the terms of the Eclipse Public License 2.0
+ * which is available at http://www.eclipse.org/legal/epl-v20.html,
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipsefoundation.marketplace.helper;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.ExecutionException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Static helper class for handling reactive stream results.
+ * 
+ * @author Martin Lowe
+ */
+public class StreamHelper {
+	private static final Logger LOGGER = LoggerFactory.getLogger(StreamHelper.class);
+
+	/**
+	 * Unwraps and awaits a completion stage to complete, returning the internal
+	 * result when finished.
+	 * 
+	 * @param <T>   the type of object that contains results of async call
+	 * @param stage the completion stage to await results for
+	 * @return the result, or null if one couldn't be found
+	 */
+	public static <T> T awaitCompletionStage(CompletionStage<T> stage) {
+		CompletableFuture<T> future = stage.toCompletableFuture();
```

Comment:
a CompletionStage is a CompletableFuture implementation, so no need to do that here.

Reply:
CompletionStage is the interface, and Quarkus also has implementations of the class that aren't straight casts which could cause issues.